src/coredump/coredump.c

```diff
@@ -68,28 +68,57 @@
 assert_cc(JOURNAL_SIZE_MAX <= DATA_SIZE_MAX);
 
 enum {
-        /* We use this as array indexes for a couple of special fields we use for naming
-         * coredump files, and attaching xattrs, and for indexing argv[].
+        /* We use these as array indexes for our process metadata cache.
          *
-         * In the internal context[] array, fields before CONTEXT_COMM are the strings
-         * from argv[] passed by the kernel according to our pattern defined in
-         * /proc/sys/kernel/core_pattern (see man:core(5)). So they should not be
-         * freed. The strings at indices CONTEXT_COMM and higher are allocated by us and
-         * should be freed at the end.
-         */
-        CONTEXT_PID,
-        CONTEXT_UID,
-        CONTEXT_GID,
-        CONTEXT_SIGNAL,
-        CONTEXT_TIMESTAMP,
-        CONTEXT_RLIMIT,
-        CONTEXT_HOSTNAME,
-        CONTEXT_COMM,
-        CONTEXT_EXE,
-        CONTEXT_UNIT,
-        _CONTEXT_MAX
+         * The first indices of the cache stores the same metadata as the ones passed by
+         * the kernel via argv[], ie the strings array passed by the kernel according to
+         * our pattern defined in /proc/sys/kernel/core_pattern (see man:core(5)). */
+
+        META_ARGV_PID,          /* %P: as seen in the initial pid namespace */
+        META_ARGV_UID,          /* %u: as seen in the initial user namespace */
+        META_ARGV_GID,          /* %g: as seen in the initial user namespace */
+        META_ARGV_SIGNAL,       /* %s: number of signal causing dump */
+        META_ARGV_TIMESTAMP,    /* %t: time of dump, expressed as seconds since the Epoch */
+        META_ARGV_RLIMIT,       /* %c: core file size soft resource limit */
+        META_ARGV_HOSTNAME,     /* %h: hostname */
+        _META_ARGV_MAX,
+
+        /* The following indexes are cached for a couple of special fields we use (and
+         * thereby need to be retrieved quickly) for naming coredump files, and attaching
+         * xattrs. Unlike the previous ones they are retrieved from the runtime
+         * environment. */
+
+        META_RT_COMM = _META_ARGV_MAX,
```

Comment:
what is RT supposed to mean?

Reply:
"runtime"

Reply:
hmm "runtime" is such a generic word, and needless abbreviation dosn't help. Maybe call this PROC, because this data is from /proc?

Reply:
That wouldn't work for META_RT_UNIT...

Maybe we could simply use META_UNIT, META_COMM, etc... so we only emphasize the fact that some metadata (META_ARGV_*) are passed by the kernel via args when it calls the coredump handler and the others are collected by the handler itself by reading /proc or by any other means ?

Reply:
the unit info comes from /proc/ too (it's parsed from /proc/$PID/cgroup), hence i think that name applies too...

but yeah, dropping the middle part of the string sounds ok to me too